pkg/util/kube/utils.go

```diff
@@ -107,7 +114,12 @@ func EnsureNamespaceExistsAndIsReady(namespace *corev1api.Namespace, client core
 	// err will be set if we timed out or encountered issues retrieving the namespace,
 	if err != nil {
 		if terminatingNamespace {
+			// If the namespace is marked for deletion, and we timed out, adding it in tracker
+			resourceDeletionStatusTracker.Add(namespace.Kind, namespace.Name, namespace.Name)
 			return false, nsCreated, errors.Wrapf(err, "timed out waiting for terminating namespace %s to disappear before restoring", namespace.Name)
+		} else if namespaceAlreadyInDeletionTracker {
+			// If the namespace is already in the tracker, return an error.
+			return false, nsCreated, err
```

Comment:
let's make this error more verbose. 
Something like - Skipping polling for terminating namespace.